### DISCUSSION TITLE
can not install booknlp

### DISCUSSION THREAD

**Discussion by ZeroCrossblade** [DrewThomasson/VoxNovel#18] [General]
i have trouble installing booknip "pip install booknlp" i on window 11 using WSL i getting this error  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for tokenizers
Failed to build tokenizers
ERROR: Could not build wheels for tokenizers, which is required to install pyproject.toml-based projects

**Comment by DrewThomasson** [comment]
Did you run all the commands in order?

I don't really know much about any issues arising like that from booknlp but try these and see if it fixes it on your end 


```bash
pip install --upgrade pip setuptools wheel
sudo apt-get update
sudo apt-get install build-essential
pip install tokenizers
#then try installing booknlp again
pip install booknlp
```



## option #2

-You could also try whipping the wsl env and use this command to install everything instead, 
-Its a single command full install for ubuntu that works in wsl on my tests from the readme
<details>


<summary> Easy Windows 11 install </summary>


  
1. In your PowerShell, paste `wsl --install` to install WSL.
2.  In your PowerShell, paste `wsl --update --pre-release` to update WSL to a pre-release version, right now the prerelease has all the sound and such working.

 
3. After you set the username and passsudo apt install nvidia-cuda-toolkitword, you'll now be able to open WSL as an application. This will allow you to run Ubuntu software. You'll now want to open WSL if you closed it and paste these commands.
4. run this command in the Ubuntu wsl for a single command install `yes | wget -O - https://raw.githubusercontent.com/DrewThomasson/VoxNovel/main/shell_install_scripts/Ubuntu-install.sh | bash`
5. Congrates! VoxNovel should run in WSL now!
</details>


Let me know if any of that worked!

**Reply by ZeroCrossblade** [reply]
i receiving those error  ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
styletts2 0.1.6 requires gruut<3.0.0,>=2.3.4, but you have gruut 2.2.3 which is incompatible.
styletts2 0.1.6 requires librosa<0.11.0,>=0.10.1, but you have librosa 0.10.0 which is incompatible.

ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
tts 0.21.3 requires transformers>=4.33.0, but you have transformers 4.30.0 which is incompatible.
styletts2 0.1.6 requires gruut<3.0.0,>=2.3.4, but you have gruut 2.2.3 which is incompatible.
styletts2 0.1.6 requires librosa<0.11.0,>=0.10.1, but you have librosa 0.10.0 which is incompatible.
styletts2 0.1.6 requires transformers<5.0.0,>=4.36.0, but you have transformers 4.30.0 which is incompatible.

ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
tts 0.21.3 requires trainer>=0.0.32, but you have trainer 0.0.31 which is incompatible.
tts 0.21.3 requires transformers>=4.33.0, but you have transformers 4.30.0 which is incompatible.
google-cloud-storage 2.16.0 requires google-auth<3.0dev,>=2.26.1, but you have google-auth 2.23.4 which is incompatible.
langchain-core 0.1.52 requires packaging<24.0,>=23.2, but you have packaging 23.1 which is incompatible.
styletts2 0.1.6 requires accelerate<0.26.0,>=0.25.0, but you have accelerate 0.24.1 which is incompatible.
styletts2 0.1.6 requires einops<0.8.0,>=0.7.0, but you have einops 0.6.1 which is incompatible.
styletts2 0.1.6 requires filelock<3.13,>=3.12.4, but you have filelock 3.13.1 which is incompatible.
styletts2 0.1.6 requires gruut<3.0.0,>=2.3.4, but you have gruut 2.2.3 which is incompatible.
styletts2 0.1.6 requires librosa<0.11.0,>=0.10.1, but you have librosa 0.10.0 which is incompatible.
styletts2 0.1.6 requires torch<3.0.0,>=2.1.2, but you have torch 2.1.0 which is incompatible.
styletts2 0.1.6 requires torchaudio<3.0.0,>=2.1.2, but you have torchaudio 2.1.0 which is incompatible.
styletts2 0.1.6 requires tqdm<5.0.0,>=4.66.1, but you have tqdm 4.64.1 which is incompatible.
styletts2 0.1.6 requires transformers<5.0.0,>=4.36.0, but you have transformers 4.30.0 which is incompatible.
styletts2 0.1.6 requires typing-extensions<5.0.0,>=4.9.0, but you have typing-extensions 4.8.0 which is incompatible.
typeguard 4.2.1 requires typing-extensions>=4.10.0; python_version < "3.13", but you have typing-extensions 4.8.0 which is incompatible.

**Reply by DrewThomasson** [reply]
You're not trying to do all the pip installs in a single command are you?

How are you installing this

**Reply by DrewThomasson** [reply]
Also if gave me a full a log of the terminal session for me to review it would be great, the issue might be something from before that you haven't pasted idk.

For these kinds of things your main worry shouldn't be giving me information overload lol


Also is this a new error that's not letting the program run after we fixed the one from before?

**Reply by ZeroCrossblade** [reply]
i doing the single command

**Reply by DrewThomasson** [reply]
The 

'yes | wget -O - https://raw.githubusercontent.com/DrewThomasson/VoxNovel/main/shell_install_scripts/Ubuntu-install.sh | bash

'
?

Just verifying lol

**Reply by ZeroCrossblade** [reply]
yes

**Reply by DrewThomasson** [reply]
And when you go to the voxnovel dir in that wsl and run 'python gui_run.py' what error do you get?

Summary:

'cd VoxNovel'
'python gui_run.py' 


-also at the same time I'm trying out the command on a fresh wsl env on a AMD windows laptop to see if I get your error

**Reply by ZeroCrossblade** [reply]
Command 'python' not found, did you mean:
  command 'python3' from deb python3
  command 'python' from deb python-is-python3
zero@Gamingdesktop:~/VoxNovel$

**Reply by ZeroCrossblade** [reply]
thank for the help im totally new to this

**Reply by DrewThomasson** [reply]
Type in... 

'conda'

?

Hm that's really weird

**Reply by ZeroCrossblade** [reply]
zero@Gamingdesktop:~/VoxNovel$ python3 gui_run.py
Traceback (most recent call last):
  File "/home/zero/VoxNovel/gui_run.py", line 25, in <module>
    import pandas as pd
ModuleNotFoundError: No module named 'pandas'

**Reply by ZeroCrossblade** [reply]
conda?

**Reply by ZeroCrossblade** [reply]
zero@Gamingdesktop:~/VoxNovel$ conda
conda: command not found

**Reply by DrewThomasson** [reply]
Yeah it should be installed with that single command,

You did reset the wsl env before doing the single command right?

**Reply by DrewThomasson** [reply]
I can show you how on windows, if you don't know exactly what I mean

**Reply by ZeroCrossblade** [reply]
yes can you show me

**Reply by DrewThomasson** [reply]
Close out of the existing wsl window 


Open powershell and type in 

'wsl --list --verbose'

You should see a Ubuntu in the list

Then type in 

'wsl --unregister Ubuntu'


That should remove it 

Then type in 

'wsl --install'

That should reinstall Ubuntu as a refresh install in wsl


Then try the single command again

**Reply by ZeroCrossblade** [reply]
after it finish installing do i close the window and open back up

**Reply by ZeroCrossblade** [reply]
(base) zero@Gamingdesktop:~$ ls
VoxNovel  miniconda
(base) zero@Gamingdesktop:~$ cd VoxNovel
(base) zero@Gamingdesktop:~/VoxNovel$ ls
1CPU_Book_processing.py        SteamDeck_Requirments.txt    linux_requirements.txt
2GPU_Audio_generation.py       Ubuntu_requirements.txt      readme.md
Colab_Headless_VoxNovel.ipynb  Windows-requirements.txt     readme_files
DockerFiles                    auto_noGui_run.py            shell_install_scripts
Example_Working_files.zip      convert⁄fast_voice_clone.py  steamdeck_Distrobox_install_guide.txt
LICENSE                        gui_run.py                   test.py
MAC-requirements.txt           headless_voxnovel.py         tortoise
(base) zero@Gamingdesktop:~/VoxNovel$ python gui_run.py
Traceback (most recent call last):
  File "/home/zero/VoxNovel/gui_run.py", line 25, in <module>
    import pandas as pd
ModuleNotFoundError: No module named 'pandas'

**Reply by DrewThomasson** [reply]
Ok 

'pip install pandas'

**Reply by DrewThomasson** [reply]
Any module not found try doing a pip install for it

**Reply by ZeroCrossblade** [reply]
it talk me to install booknlp do i need to install anything else before that

**Reply by DrewThomasson** [reply]
No? But that is weird idk why your computer is acting up on wsl

Your getting too many modual errors

What happens when you type in

'conda' 

now?

And 

'conda env list'

**Reply by DrewThomasson** [reply]
It's almost as if your not in the voxnovel conda env is why I'm confused

**Reply by DrewThomasson** [reply]
Yeah you might not be in the VoxNovel conda env that holds all of the pips that the single command did


Try 

'conda env list'

You should see a voxnovel in there

And then 

'conda activate VoxNovel'

And if that works then try re running the gui_run.py file with 

'python gui_run.py'

**Reply by ZeroCrossblade** [reply]
it started thank for the help do i do this again when i turn on the terminal

**Reply by DrewThomasson** [reply]
Omg yes yes!

You know how to access the output files of it in windows also right?

**Reply by ZeroCrossblade** [reply]
\\wsl.localhost\Ubuntu\home\

**Reply by DrewThomasson** [reply]
It's in the readme but just making sure you should be able to access all the files of the Ubuntu wsl machine in your file explorer by going to 


\\wsl.localhost\ubuntu\home\

Or you should see the little Linux guy in the side bar in the file explorer 

And it should be in the VoxNovel folder in a folder called 'output_audiobooks'

The details for the folders are in the "Folders" section of the readme


Good Luck!
And no probs glad to see it working on your end!

**Reply by DrewThomasson** [reply]
Oh good u got it nice!

**Reply by ZeroCrossblade** [reply]
ok thank im new to terminal uses
